test/cljam/algo/pileup_test.clj

```diff
@@ -198,6 +198,26 @@
                                   [{:pos 3 :pile [{:pos 3 :end 3}]}]
                                   [])))))
 
+(deftest chunk-seq
+  (testing "empty"
+    (is (nil? (seq (#'plp/chunk-seq 1 10 2 []))))
+    (is (nil? (seq (#'plp/chunk-seq 2 10 2 (map ->aln [{:pos 1 :end 1}])))))
+    (is (nil? (seq (#'plp/chunk-seq 1 10 2 (map ->aln [{:pos 11 :end 11}]))))))
+  (testing "dense"
+    (are [?in ?out]
+        (= ?out (mapv ->pile (#'plp/chunk-seq 1 10 2 (mapv ->aln (map #(zipmap [:pos :end] %) ?in)))))
+      [[1 1]] [[1 [[1 1]]]]
+      [[3 5]] [[3 [[3 5]]]]
+      [[3 5] [4 4]] [[3 [[3 5] [4 4]]]]
+      [[3 5] [4 6]] [[3 [[3 5] [4 6]]] [6 [[4 6]]]]))
+  (testing "sparse"
+    (are [?in ?out]
+        (= ?out (mapv ->pile (#'plp/chunk-seq 1 10 2 (mapv ->aln (map #(zipmap [:pos :end] %) ?in)))))
+      [[3 5] [14 16]] [[3 [[3 5]]]]
+      [[3 5] [6 7]] [[3 [[3 5]]] [6 [[6 7]]]]
+      [[3 5] [4 5] [6 7]] [[3 [[3 5] [4 5]]] [6 [[6 7]]]]
+      [[3 5] [7 8] [10 11]] [[3 [[3 5]]] [7 [[7 8]]] [10 [[10 11]]]])))
```

Comment:
Please add cases for some large chunk sizes like `#'plp/chunk-seq 1 10 20`

Reply:
Added some patterns